src/GNDStk/Node/src/call.hpp

```diff
@@ -41,17 +52,20 @@ decltype(auto) operator()(
 ) GNDSTK_CONST {
    bool &found = detail::extract_found(std::forward<KEYWORDS>(keywords)...);
    try {
+      // ""?
       if (kwd.name == "")
          detail::apply_converter<TYPE>{}(kwd,*this);
-      return child(-kwd.one(), detail::label_is(label), found)
-                  (std::forward<KEYWORDS>(keywords)...);
+      // total filter
+      auto filter = [kwd,label](const Node &n)
+         { return kwd.filter(n) && detail::label_is(label)(n); };
+      // -(--kwd): child_t<void,allow::one,...>
+      return child(-(--kwd)+filter, found)(std::forward<KEYWORDS>(keywords)...);
```

Comment:
OK, so I know that --kwd degrades the many state of the keyword to one, but what does the negation of the result do?

Reply:
There's a system to these operators. :-)

ONE minus (unary minus operator) downgrades the FIRST template argument.
TWO minuses (predecrement operator) downgrades the SECOND template argument.

So, one minus takes a `meta_t<type,...>` or `child_t<type,...>` and downgrades to `meta_t<void>` or `child_t<void>`.
Which, as you know, means to retrieve the `meta_t` or `child_t` in their raw tree form: const or non-const string or node reference.

Here's what you're really seeing in the code block you referenced. We're traversing a query sequence, say `node(A,B,C,D,E)`. The code does this by "peeling off" the first argument in order to get to the child node (or the current `*this` node) represented by the first argument. It downgrades to `<void>`, even if the `child_t` has a type, because we need to stay in tree form (so we can continue going to child nodes) until we get to the very last argument - E, in my example. In this way, non-`<void>` `child_t` objects can appear in sequence, without the user having to explicitly downgrade them just because they're not at the end.

As for the `--` in the referenced code, it's because - in that variation of `operator()` - we have something like `node(...,child_t,"foobar",...)`, for which GNDStk's interpretation is to look for the specific case of the child node that has metadatum `"label" == "foobar"`. For that interpretation, GNDStk allows even an `allow::many` `child_t` (which normally wouldn't be allowed before the end of the sequence), because it assumes that the extra label-match condition means there's probably just one, not many.

Does that make sense?

Reply:
A bit of history: the unary minus is something I put in a long time ago, well before all the other operators, because I needed to do just what I described: stay in `<void>` form - raw tree form - until we get to the end.

The original thinking was: I'm calling this my "smart keyword" system, because it can deal with custom user types.

A "dumb keyword", on the other hand, would just have the `<void>` meaning - stay in tree form, without the nice type system.

So, unary MINUS takes a smart keyword and "SUBTRACTS from its intelligence", making it a dumb keyword. :-p